kivy/uix/widget.py

```diff
@@ -660,6 +660,73 @@ def export_to_png(self, filename, *args, **kwargs):
 
         return True
 
+    def export_to_base64(self, *args, **kwargs):
+        '''Saves an image of the widget and its children in base64 format saves into memory instead of storing it.
+        Works by removing the widget canvas from its
+        parent, rendering to an :class:`~kivy.graphics.fbo.Fbo`, and calling
+        :meth:`~kivy.graphics.texture.Texture.save`.
+
+        .. note::
+
+            The image includes only this widget and its children. If you want
+            to include widgets elsewhere in the tree, you must call
+            :meth:`~Widget.export_to_base64` from their common parent, or use
+            :meth:`~kivy.core.window.WindowBase.screenshot` to capture the
+            whole window.
+
+        .. note::
+
+            The image will be returned in a base64 format.
+
+        .. versionadded:: 1.9.0
+
+        :Parameters:
+            `method`: str
+                empty or 'web'. Note: None is for returning the image in Kivy and to be shown.
+            `scale`: float
+                The amount by which to scale the saved image, defaults to 1.
+
+                .. versionadded:: 1.11.0
+        '''
+        try:
+            from PIL import Image
+        except ImportError as error:
+            raise Exception(error)
```

Comment:
Why are we rerasing to a less explicit type? Why even try/catching?

Reply:
I wanted it to look good. And a few Google searches this seemed to be a good one to use.

Reply:
The idea is reraising needs to serve a purpose, like specializing the exception or logging info prior to raising or something.
If you find that link you came across in your Google searches, feel free to share it, I'm curious about it. I could be missing something

Reply:
Here it is (Stackoverflow)[https://stackoverflow.com/a/3131251]

Reply:
OK so here what they do is to fallback on another import if one doesn't exist, that's the reason they try/except. They don't try/except to just reraise, you see what I mean?
For instance this pattern is used for urllib import that works with both Python2 and Python3:
```python
try:
    from urllib.parse import urlparse
except ImportError:
    from urlparse import urlparse
```
Another example could be to import a non-mandatory module:
```python
try:
    import foo
except ImportError:
    foo = None

# then later in the code
if foo is not None:
    foo.whatever()
```
But again there's no reraise. One reason to reraise could be:
```python
try:
    import foobar
except ImportError:
    logger.error('I want dedicated logs for this error')
    raise
```
Or:
try:
    import foobar
except ImportError:
    raise FoobarModuleNotFound('whatever message')
```
This sometimes makes sense in specific cases, e.g. if you're dealing with `FoobarModuleNotFound` later on.

Does it make more sense?

Reply:
Aa okay! But you should be able to edit this because I allowed it when I made this PR.

Reply:
addressed in daab224